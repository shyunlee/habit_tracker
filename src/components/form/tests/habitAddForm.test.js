import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";
import HabitAddForm from "../HabitAddForm";

describe("HabitAddForm", () => {
  it("Render Snapshot Test", () => {
    const component = renderer.create(<HabitAddForm onAdd={jest.fn()} />)
    expect(component.toJSON).toMatchSnapshot()
  })

  describe("Form Submit Test", () => {
    let onAdd;
    let input;
    let button;

    beforeEach(() => {
      onAdd = jest.fn();
      render(<HabitAddForm onAdd={onAdd} />);
      input = screen.getByPlaceholderText("Habit");
      button = screen.getByText("Add");
    });

    it("calls add function from props when name is input and valid", () => {
      userEvent.type(input, "New Habit");
      userEvent.click(button);

      expect(onAdd).toHaveBeenCalledWith("New Habit");
    });

    it("shold not call add function from props when name is empty", () => {
      userEvent.type(input, "");
      userEvent.click(button);

      expect(onAdd).toHaveBeenCalledTimes(0);
    });
  });
});
