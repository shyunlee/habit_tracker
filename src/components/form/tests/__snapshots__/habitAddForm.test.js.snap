// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HabitAddForm Render Snapshot Test 1`] = `[Function]`;
